src/options/mkoptions.py

```diff
@@ -0,0 +1,1428 @@
+                        TPL_CALL_ASSIGN.format(
+                            name=option.name,
+                            option='"{}"'.format(smtname)))
+                elif option.handler:
+                    h = 'handler->{handler}("{smtname}"'
+                    if argument_req:
+                        h += ', optionarg'
+                    h += ');'
+                    setoption_handlers.append(
+                        h.format(handler=option.handler, smtname=smtname))
+
+                if smtlinks:
+                    setoption_handlers.append('\n'.join(smtlinks))
+                setoption_handlers.append('return;')
+                setoption_handlers.append('}')
+
+                if option.name:
+                    getoption_handlers.append(
+                        'if ({}) {{'.format(cond))
+                    if option.type == 'bool':
+                        getoption_handlers.append(
+                            'return options::{}() ? "true" : "false";'.format(
+                                option.name))
+                    else:
+                        getoption_handlers.append('std::stringstream ss;')
+                        if is_numeric_cpp_type(option.type):
+                            getoption_handlers.append(
+                                'ss << std::fixed << std::setprecision(8);')
+                        getoption_handlers.append('ss << options::{}();'.format(
+                            option.name))
+                        getoption_handlers.append('return ss.str();')
+                    getoption_handlers.append('}')
+
+
+            # Add --no- alternative options for boolean options
+            if option.long and option.type == 'bool' and option.alternate:
+                cases = []
+                cases.append(
+                    'case {}:// --no-{}'.format(
+                        g_getopt_long_start + len(getopt_long),
+                        option.long))
+                cases.append(
+                    TPL_CALL_ASSIGN_BOOL.format(
+                        name=option.name, option='option', value='false'))
+                cases.append('  break;\n')
+
+                options_handler.extend(cases)
+
+                add_getopt_long('no-{}'.format(option.long), argument_req,
+                                getopt_long)
+
+            optname = option.smt_name if option.smt_name else option.long
+            # collect options available to the SMT-frontend
+            if optname:
+                options_smt.append('"{}",'.format(optname))
+
+            if option.name:
+                # Build options for options::getOptions()
+                if optname:
+                    # collect SMT option names
+                    options_smt.append('"{}",'.format(optname))
+
+                    if option.type == 'bool':
+                        s = '{ std::vector<std::string> v; '
+                        s += 'v.push_back("{}"); '.format(optname)
```

Comment:
We should use `emplace_back` here in the future. Right now the goal is to generate code that is similar to the old one, so probably needs to be fixed in a separate PR?

Reply:
We have C++11, this is a good candidate for an initializer list:
opts.push_back(std::vector<string>{$optname, $option.name ? ... }).

Reply:
Yes, we should address comments regarding the generated code in a separate PR.